src/hooks/useConnectionStatus.tsx

```diff
@@ -40,12 +33,35 @@ export const ConnectionStatusProvider: React.FC = ({ children }) => {
         4000
       )
       if (res) {
-        setIsUserConnectedToInternet(true)
+        setIsUserConnectedToInternet((previousState) => {
+          if (previousState === false) {
+            setShowSnackbar(true)
+          }
+          return true
+        })
       }
     } catch (error) {
       setIsUserConnectedToInternet(false)
+      displaySnackbar({ title: 'No network connection', iconType: 'error' })
     }
-  }
+  }, [displaySnackbar])
+
+  useEffect(() => {
+    // ping google every three seconds to check if user is connected to internet
```

Comment:
```suggestion
    // ping google every five seconds to check if user is connected to internet
```
:P but I guess we could just remove this comment